schemaPoolDocuments.go

```diff
@@ -0,0 +1,48 @@
+package gojsonschema
+
+import "sync"
+
+type schemaPoolDocuments struct {
+	mp sync.Map
+}
+
+func (spd *schemaPoolDocuments) Load(key string) (*schemaPoolDocument, bool) {
+	val, ok := spd.mp.Load(key)
```

Comment:
tbh, I wonder if the added complexity of using a `sync.Map` is needed for this use-case. `sync.Map` (by itself) does not have type-safety, so requires all the extra boilerplating in this file, and while `sync.Map` _can_ perform better, in many situations it _may_ actually be slower (unless _high_ concurrency, large number of cores); https://medium.com/@deckarep/the-new-kid-in-town-gos-sync-map-de24a6bf7c2c